backends/bmv2/midend.cpp

```diff
@@ -122,6 +123,11 @@ MidEnd::MidEnd(BMV2Options& options) {
         new P4::CompileTimeOperations(),
         new P4::TableHit(&refMap, &typeMap),
         new P4::RemoveLeftSlices(&refMap, &typeMap),
+
+        // p4c-bm removed unused action parameters. To produce a compatible
+        // control plane API, we remove them as well for P4-14 programs.
+        isv1 ? new P4::RemoveUnusedActionParameters(&refMap) : nullptr,
```

Comment:
We have been trying to remove all differences between P4-14 and P4-16 compilation. This flag means that if you convert a p4-14 program to p4-16 then the compilation will produce a different result than if you start with just p4-14.
So it looks to me like what you want is a new option flag instead; "--removeUnusedActionArgs", which is provided for compatibility with some old tools.